src/openvic-simulation/modifier/ModifierManager.cpp

```diff
@@ -35,466 +56,517 @@ bool ModifierManager::add_modifier_effect(
+	std::string_view localisation_key,
+	bool has_no_effect
+) { return _register_modifier_effect(leader_modifier_effects, ModifierEffect::target_t::UNIT, effect_cache, std::move(identifier), is_positive_good, format, localisation_key, has_no_effect); }
+bool ModifierManager::register_unit_terrain_modifier_effect(
+	ModifierEffect const*& effect_cache,
+	std::string_view identifier,
+	bool is_positive_good,
+	ModifierEffect::format_t format,
+	std::string_view localisation_key,
+	bool has_no_effect
+) { return _register_modifier_effect(unit_terrain_modifier_effects, ModifierEffect::target_t::COUNTRY, effect_cache, std::move(identifier), is_positive_good, format, localisation_key, has_no_effect); }
+bool ModifierManager::register_shared_tech_country_modifier_effect(
+	ModifierEffect const*& effect_cache,
+	std::string_view identifier,
+	bool is_positive_good,
+	ModifierEffect::format_t format,
+	std::string_view localisation_key,
+	bool has_no_effect
+) { return _register_modifier_effect(shared_tech_country_modifier_effects, ModifierEffect::target_t::COUNTRY, effect_cache, std::move(identifier), is_positive_good, format, localisation_key, has_no_effect); }
+bool ModifierManager::register_technology_modifier_effect(
+	ModifierEffect const*& effect_cache,
+	std::string_view identifier,
+	bool is_positive_good,
+	ModifierEffect::format_t format,
+	std::string_view localisation_key,
+	bool has_no_effect
+) { return _register_modifier_effect(technology_modifier_effects, ModifierEffect::target_t::COUNTRY, effect_cache, std::move(identifier), is_positive_good, format, localisation_key, has_no_effect); }
+bool ModifierManager::register_base_country_modifier_effect(
+	ModifierEffect const*& effect_cache,
+	std::string_view identifier,
+	bool is_positive_good,
+	ModifierEffect::format_t format,
+	std::string_view localisation_key,
+	bool has_no_effect
+) { return _register_modifier_effect(base_country_modifier_effects, ModifierEffect::target_t::COUNTRY, effect_cache, std::move(identifier), is_positive_good, format, localisation_key, has_no_effect); }
+bool ModifierManager::register_base_province_modifier_effect(
+	ModifierEffect const*& effect_cache,
+	std::string_view identifier,
+	bool is_positive_good,
+	ModifierEffect::format_t format,
+	std::string_view localisation_key,
+	bool has_no_effect
+) { return _register_modifier_effect(base_province_modifier_effects, ModifierEffect::target_t::PROVINCE, effect_cache, std::move(identifier), is_positive_good, format, localisation_key, has_no_effect); }
+bool ModifierManager::register_terrain_modifier_effect(
+	ModifierEffect const*& effect_cache,
+	std::string_view identifier,
+	bool is_positive_good,
+	ModifierEffect::format_t format,
+	std::string_view localisation_key,
+	bool has_no_effect
+) { return _register_modifier_effect(terrain_modifier_effects, ModifierEffect::target_t::PROVINCE, effect_cache, std::move(identifier), is_positive_good, format, localisation_key, has_no_effect); }
```

Comment:
Formatting + unnecessary code duplication, use a macro instead (and undefine it at the end).

Reply:
I'll rewrite it using a macro. This looks like a great use case for one.

Reply:
I also rewrote the hpp using a macro.